d3a_interface/matching_algorithms/requirements_validators.py

```diff
@@ -0,0 +1,205 @@
+            f"Invalid data type for energy {requirement}"
+        assert isinstance(selected_energy, (int, float)), \
+            f"Invalid data type for selected_energy {selected_energy}"
+        # bid_required_energy is None or it is defined & equals selected_energy -> true
+        return bid_required_energy >= selected_energy
+
+
+class PriceRequirement(Requirement):
+    """Check if price requirement of bid is satisfied."""
+
+    @classmethod
+    def is_satisfied(cls, offer: Offer, bid: Bid, requirement: Dict,
+                     clearing_rate: Optional[float] = None,
+                     selected_energy: Optional[float] = None) -> bool:
+        bid_required_price = requirement.get("price")
+        assert isinstance(bid_required_price, (int, float)), \
+            f"Invalid data type for energy {requirement}"
+        assert isinstance(selected_energy, (int, float)), \
+            f"Invalid data type for selected_energy {selected_energy}"
+        assert isinstance(clearing_rate, (int, float)), \
+            f"Invalid data type for clearing_rate {clearing_rate}"
+        # bid_required_price <= price -> true
+        return bid_required_price <= selected_energy * clearing_rate
+
+
+# Supported offers/bids requirements
+# To add a new requirement create a class extending the Requirement abstract class and add it below
+
+SUPPORTED_OFFER_REQUIREMENTS = {
+    "trading_partners": TradingPartnersRequirement
+}
+
+SUPPORTED_BID_REQUIREMENTS = {
+    "trading_partners": TradingPartnersRequirement,
+    "energy_type": EnergyTypeRequirement,
+    "energy": SelectedEnergyRequirement,
+    "price": PriceRequirement
+}
+
+
+class RequirementsSatisfiedChecker:
+    """Check if a list of bid/offer requirements are satisfied."""
+
+    @classmethod
+    def is_satisfied(
+            cls, offer: Union[Offer, Offer.serializable_dict],
+            bid: Union[Bid, Bid.serializable_dict],
+            clearing_rate: Optional[float] = None,
+            selected_energy: Optional[float] = None) -> bool:
+        """Check whether at least 1 offer and 1 bid requirement are satisfied."""
+
+        offer = Offer.from_dict(offer) if isinstance(offer, dict) else offer
+        bid = Bid.from_dict(bid) if isinstance(bid, dict) else bid
+
+        if offer.requirements:
+            offer_requirement_satisfied = cls.are_offer_requirements_satisfied(
+                offer, bid, clearing_rate, selected_energy)
+        else:
+            # If concerned offer have no requirements, consider it as satisfied.
+            offer_requirement_satisfied = True
+
+        if bid.requirements:
+            bid_requirement_satisfied = cls.are_bid_requirements_satisfied(
+                offer, bid, clearing_rate, selected_energy)
+        else:
+            # If concerned bid have no requirements, consider it as satisfied.
+            bid_requirement_satisfied = True
+
+        return offer_requirement_satisfied and bid_requirement_satisfied
```

Comment:
Nit, but since in this code we're differentiating between individual and total requirements:
```suggestion
        return offer_requirements_satisfied and bid_requirements_satisfied
```